### PR TITLE
Fix: Wallets being marked as backed up by walletLoadState()

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1296

### CODE REVIEW

File: src/components/backup/RestoreCloudStep.tsx [hunk 1]
```diff
@@ -130,34 +145,51 @@ export default function RestoreCloudStep() {
...
+          // @ts-expect-error TypeScript doesn't play nicely with Redux types here
+          const walletIdsToUpdate = Object.keys(newWalletsState || {}).filter(walletId => !(prevWalletsState || {})[walletId]);
           logger.log('updating backup state of wallets with ids', {
             walletIds: JSON.stringify(walletIdsToUpdate),
           });
           logger.log('backupSelected.name', {
             fileName: selectedBackup.name,
           });
+
+          console.log(JSON.stringify(walletIdsToUpdate, null, 2));
```
Comment (walmat) [inline src/components/backup/RestoreCloudStep.tsx:171]:
```suggestion
```

File: src/components/backup/RestoreCloudStep.tsx [hunk 2]
```diff
@@ -130,34 +145,50 @@ export default function RestoreCloudStep() {
 
       if (status === RestoreCloudBackupResultStates.success) {
         // Store it in the keychain in case it was missing
-        await saveBackupPassword(password);
+        const hasSavedPassword = await getLocalBackupPassword();
+        if (!hasSavedPassword) {
+          await saveLocalBackupPassword(password);
+        }
```
Comment (walmat) [inline src/components/backup/RestoreCloudStep.tsx:151]: @skylarbarrera does this make sense to do here? This is the new local backup function we wrote not sure if you remember.

File: src/components/backup/useCreateBackup.ts [hunk 3]
```diff
@@ -49,7 +48,8 @@ export const useCreateBackup = ({ walletId, navigateToRoute }: UseCreateBackupPr
     [setLoading]
   );
   const onSuccess = useCallback(async () => {
-    if (!alreadyHasLocalPassword) {
+    const hasSavedPassword = await getLocalBackupPassword();
+    if (!hasSavedPassword) {
```
Comment (walmat) [inline src/components/backup/useCreateBackup.ts:52]: @skylarbarrera also think the main issue here was the state variable wasn't be persisted when we were fetching the password on the BackupCloudStep component.

File: src/components/backup/RestoreCloudStep.tsx [hunk 4]
```diff
@@ -122,6 +135,8 @@ export default function RestoreCloudStep() {
         throw new Error('No backup file selected');
       }
 
+      const prevWalletsState = await dispatch(walletsLoadState());
```
[the remark this answers is not in the document]
Comment (skylarbarrera) [reply src/components/backup/RestoreCloudStep.tsx:138]: didnt think this would return

File: src/components/backup/RestoreCloudStep.tsx (hunk 2 above)
Comment (skylarbarrera) [reply src/components/backup/RestoreCloudStep.tsx:151]: this seems fine, i think the issue was that password could be undefined/empty? that issue seems diff

File: src/components/backup/useCreateBackup.ts (hunk 3 above)
Comment (skylarbarrera) [reply src/components/backup/useCreateBackup.ts:52]: yeah i think that makes sense to me

File: src/components/backup/RestoreCloudStep.tsx [hunk 5]
```diff
@@ -130,34 +145,50 @@ export default function RestoreCloudStep() {
...
+            // NOTE: This handles cloud and manual backups
+            if (wallet.backedUp && wallet.backupType === walletBackupTypes.cloud) {
+              oldCloudIds.push(wallet.id);
+            } else if (wallet.backedUp && wallet.backupType === walletBackupTypes.manual) {
+              oldManualIds.push(wallet.id);
+            }
+          });
+
+          await dispatch(setAllWalletsWithIdsAsBackedUp(oldCloudIds, walletBackupTypes.cloud, filename));
+          await dispatch(setAllWalletsWithIdsAsBackedUp(oldManualIds, walletBackupTypes.manual, filename));
```
[the remark this answers is not in the document]
Comment (skylarbarrera) [reply src/components/backup/RestoreCloudStep.tsx:187]: this hurts my head, may need to huddle on monday

File: src/components/backup/RestoreCloudStep.tsx [hunk 6]
```diff
@@ -130,34 +145,50 @@ export default function RestoreCloudStep() {
...
+          const walletIdsToUpdate = Object.keys(newWalletsState || {}).filter(walletId => !(prevWalletsState || {})[walletId]);
           logger.log('updating backup state of wallets with ids', {
             walletIds: JSON.stringify(walletIdsToUpdate),
           });
           logger.log('backupSelected.name', {
             fileName: selectedBackup.name,
           });
+
           await dispatch(setAllWalletsWithIdsAsBackedUp(walletIdsToUpdate, walletBackupTypes.cloud, filename));
+          // end mark
```
Comment (skylarbarrera) [inline src/components/backup/RestoreCloudStep.tsx:172]: deleet

**Review by skylarbarrera — APPROVED** [review]

**Pull request merged**